processing/src/main/java/org/apache/druid/segment/QueryableIndexColumnSelectorFactory.java

```diff
@@ -76,21 +77,26 @@
   @Override
   public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec)
   {
-    return dimensionSelectorCache.computeIfAbsent(
-        dimensionSpec,
-        spec -> {
-          if (virtualColumns.exists(spec.getDimension())) {
-            DimensionSelector dimensionSelector = virtualColumns.makeDimensionSelector(dimensionSpec, index, offset);
-            if (dimensionSelector == null) {
-              return virtualColumns.makeDimensionSelector(dimensionSpec, this);
-            } else {
-              return dimensionSelector;
-            }
-          }
-
-          return spec.decorate(makeDimensionSelectorUndecorated(spec));
+    Function<DimensionSpec, DimensionSelector> mappingFunction = spec -> {
+      if (virtualColumns.exists(spec.getDimension())) {
+        DimensionSelector dimensionSelector = virtualColumns.makeDimensionSelector(dimensionSpec, index, offset);
+        if (dimensionSelector == null) {
+          return virtualColumns.makeDimensionSelector(dimensionSpec, this);
+        } else {
+          return dimensionSelector;
         }
-    );
+      }
+
+      return spec.decorate(makeDimensionSelectorUndecorated(spec));
+    };
+
+    DimensionSelector dimensionSelector = dimensionSelectorCache.get(dimensionSpec);
```

Comment:
There should be a comment explaining why it's not computeIfAbsent(), e. g. referencing back to this PR.

Reply:
done, added a comment